tests/test_cms_menus.py

```diff
@@ -404,6 +405,239 @@ def test_menu_without_softroots(self):
+                AttributeObject(title=aaa.title, level=1, children=[]),
+                AttributeObject(title=bbb.title, level=1, children=[])
+            ])
+        ]
+
+        self.assertTreeQuality(hard_root, mock_tree, 'level', 'title')
+
+    def test_menu_with_rendering_hidden_softroot(self):
+        """
+        Tree in fixture :
+               root
+                   aaa( soft_root is True and hide_noe is True)
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
+                0:aaa
+                    1:aaa1
+                        2:ccc
+                            3:ddd
+                    1:aaa2
+        Note: If node has in_navigation False , node and its descendants are removed from tree
+        """
+        menu_content = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        root = factories.ChildMenuItemFactory(parent=menu_content.root, content=self.root_pagecontent.page)
+        aaa = factories.ChildMenuItemFactory(
+            parent=root,
+            content=self.aaa_pagecontent.page,
+            soft_root=True,
+            hide_node=True,
+        )
+        aaa1 = factories.ChildMenuItemFactory(parent=aaa, content=self.aaa1_pagecontent.page)
+        ccc = factories.ChildMenuItemFactory(parent=aaa1, content=self.ccc_pagecontent.page)
+        ddd = factories.ChildMenuItemFactory(parent=ccc, content=self.ddd_pagecontent.page)
+        aaa2 = factories.ChildMenuItemFactory(parent=aaa, content=self.aaa2_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=root, content=self.bbb_pagecontent.page)
+
+        page = self.aaa_pagecontent.page
+        context = self.get_context(page.get_absolute_url(), page=page)
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        hard_root = context['children']
+
+        mock_tree = [
+            AttributeObject(title=aaa.title, level=0, children=[
+                AttributeObject(title=aaa1.title, level=1, children=[
+                    AttributeObject(title=ccc.title, level=2, children=[
+                        AttributeObject(title=ddd.title, level=3, children=[])
+                    ])
+                ]),
+                AttributeObject(title=aaa2.title, level=1, children=[])
+            ])
+        ]
+
+        self.assertTreeQuality(hard_root, mock_tree, 'level', 'title')
+
+    def test_menu_with_rendering_child_node_of_hidden_softroot(self):
+        """
+        Tree in fixture :
+               root
+                   aaa( soft_root is True and hide_noe is True)
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
```

Comment:
I don't understand why this is the expected result. Please add more detail to the description to help me understand what we are trying to test here. If I had to repair this in the future I would have no idea.